beta-pipelines/multiome/atac.wdl

```diff
@@ -22,72 +23,129 @@ workflow ATAC {
   }
 
   parameter_meta {
-    fastq_gzipped_input_read1: "read 1 fastq file as input for the pipeline, the cellular barcodes must be the first part of the read name seperated by colon"
-    fastq_gzipped_input_read2: "read 2 fastq file as input for the pipeline, the cellular barcodes must be the first part of the read name separated by colon"
+    read1_fastq_barcodes_gzipped: "read 1 fastq file as input for the pipeline, the cellular barcodes must be the first part of the read name seperated by colon"
+    read2_fastq_barcodes_gzipped: "read 2 fastq file as input for the pipeline, the cellular barcodes must be the first part of the read name separated by colon"
     output_base_name: "base name to be used for the pipelines output and intermediate files"
     monitoring_script : "script to monitor resource comsumption of tasks"
     tar_bwa_reference: "the pre built tar file containing the reference fasta and cooresponding reference files for the BWA aligner"
 
   }
 
+  call AddBarcodes {
+    input:
+      read1_fastq = read1_fastq_gzipped,
+      read3_fastq = read3_fastq_gzipped,
+      barcodes_fastq = read2_fastq_gzipped,
+      output_base_name = output_base_name
+  }
   call TrimAdapters {
     input:
-      fastq_input_read1 = fastq_gzipped_input_read1,
-      fastq_input_read2 = fastq_gzipped_input_read2,
+      fastq_input_read1 = AddBarcodes.fastq_barcodes_output_read1,
+      fastq_input_read3 = AddBarcodes.fastq_barcodes_output_read3,
       output_base_name = output_base_name,
       monitoring_script = monitoring_script
-    }
-
+   }
   call BWAPairedEndAlignment {
     input:
       fastq_input_read1 = TrimAdapters.fastq_trimmed_adapter_output_read1,
-      fastq_input_read2 = TrimAdapters.fastq_trimmed_adapter_output_read2,
+      fastq_input_read3 = TrimAdapters.fastq_trimmed_adapter_output_read3,
       tar_bwa_reference = tar_bwa_reference,
       output_base_name = output_base_name,
       monitoring_script = monitoring_script
-    }
+  }
     
-    output {
-      File bam_aligned_output = BWAPairedEndAlignment.bam_aligned_output
-   }   
+  output {
+    File bam_aligned_output = BWAPairedEndAlignment.bam_aligned_output
+  }   
 }
 
+  task AddBarcodes {
+    input {
+      File read1_fastq
+      File read3_fastq
+      File barcodes_fastq
+      String output_base_name
+      Int mem_size = 200
```

Comment:
I think mem_size might be a bit too high here. The python script should be able to run with lower memory, say perhaps 5 GB.